### PR TITLE
Use params default values when creating study

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/components/dialogs/create-study-dialog/create-study-dialog.js [hunk 1]
```diff
@@ -126,6 +127,21 @@ const CreateStudyDialog = ({ open, onClose, providedExistingCase }) => {
...
+                                : param.defaultValue,
+                        })
+                    );
+
+                    const currentParametersObj = Object.assign(
+                        {},
+                        ...currentParameters
+                    );
+
+                    setValue(CURRENT_PARAMETERS, currentParametersObj);
```
Comment (ghelloumi) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js]: you can use reduce if you want

const currentParametersObj = result.parameters.reduce((obj, param) => {
  obj[param.name] = Array.isArray(param.defaultValue) ? param.defaultValue.toString() : param.defaultValue;
  return obj;
}, {});

setValue(CURRENT_PARAMETERS, currentParametersObj);

**Review by ghelloumi — APPROVED** [review]

File: src/components/dialogs/create-study-dialog/create-study-dialog.js [hunk 2]
```diff
@@ -49,6 +50,30 @@ import PrefilledNameInput from '../commons/prefilled-name-input';
 
 const STRING_LIST = 'STRING_LIST';
 
+function customizeCurrentParameters(params) {
+    return params.reduce((obj, parameter) => {
+        // we check if the parameter is for extension, if it is, we select all possible values by default.
```
Comment (etiennehomer) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js]: we check if the parameter is for extensions. If so, we select all possible values by default.

Comment (etiennehomer) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js:55]: we check if the parameter is for extensions. If so, we select all possible values by default.

Comment (etiennehomer) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js]: we check if the parameter is for extensions. If so, we select all possible values by default.

File: src/components/dialogs/create-study-dialog/create-study-dialog.js [hunk 3]
```diff
@@ -49,6 +50,30 @@ import PrefilledNameInput from '../commons/prefilled-name-input';
...
+        // the only way for the moment to check if the parameter is for extension, is by checking his name.by def
```
Comment (etiennehomer) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js]: Remove "by def" nope ?

Comment (etiennehomer) [reply src/components/dialogs/create-study-dialog/create-study-dialog.js]: And you can keep a TODO
TODO: implement a cleaner way to determine the extensions field

**Review by etiennehomer — APPROVED** [review]

**Review by sBouzols — APPROVED** [review]

**Pull request merged**